### DISCUSSION TITLE
Why there is not Body.prototype.getReader API？

### DISCUSSION THREAD

**Discussion by manooog** [node-fetch/node-fetch#1711] [Q&A] [answered]
As we can see, Body is sub of Stream,which has getReader API,see below:

https://streams.spec.whatwg.org/#ref-for-rs-get-reader%E2%91%A4

So, why node-fetch not implement the same interface?

**Comment by jimmywarting** [comment, marked as answer]
`node-fetch` was create long before web streams where introduced into NodeJS, and most node api used `node:streams` anyways which lead to us not simply using it. so we compromised a bit cuz most ppl would just convert it into a `node:stream` anyway  to use it along with `node:fs` and other things. 

Now this have changed. web streams have been introduced into NodeJS, more and more api's accept Uint8Array and async iterators now instead. So a async iterable web stream can be written to the disc without conversion back into a `node:stream` and then being piped to.

Now we could switch to using web streams, and those who still need `node:streams` could use `streams.Readable.from(iterable)` 
but that would be a very large breaking change for many ppl who depend on node-fetch today. 

but the opposite is also possible to do using `stream.Readable.toWeb(response.body)` but it also quite very new-ish. and we still need to support older LTS version that don't have this new capabilities today. We could introduce a new breaking change but then again, we could just now recommend that ppl just switch to using the new built-in fetch api provided by NodeJS themself now from v18+ 
the built in is based on web stream and follows the fetch specification more correctly. 
so our recommendation is to use the built in fetch in v18
if you do want to use it in v16 then you would have to depend on `undici` directly instead. That's the official fetch api that's implemented in NodeJS today.

**Reply by manooog** [reply]
Thanks for your reply!
